### PR TITLE
feat: update icons with new design

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes UNIFY-1105

### User facing changelog
Before

icons_before

After

icons_after

### Additional details

Once this is merged, the main repo should replace https://github.com/cypress-io/cypress-icons.

Omitted any AI or Figma assets in the logo directory as I do not know if they are public.

Deleted the tray directory and the related function since we do not seem to use it anymore.

### How has the user experience changed?
Better icons

### CODE REVIEW

File: npm/icons/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@cypress/icons",
```
Comment (flotwig) [inline npm/icons/package.json:2]: are we going to continue publishing this NPM package? does anyone else use this NPM package? if we're only using it in the monorepo we could move this to `packages` and stop having to publish this to NPM

Comment (elevatebart) [reply npm/icons/package.json:2]: We could stop, I did not think answering this question and archiving the repo + npm package would be part of this PR.

File: packages/electron/lib/install.js [hunk 2]
```diff
@@ -25,44 +23,66 @@ module.exports = {
...
+    })
+  },
+
+  checkIconVersion () {
+    const mainIconsPath = this.icons().getPathToIcon('cypress.icns')
+    const cachedIconsPath = path.join(__dirname, '../dist/Cypress/Cypress.app/Contents/Resources/electron.icns')
+
+    return Promise.all([this.getFileHash(mainIconsPath), this.getFileHash(cachedIconsPath)])
+    .then(([mainHash, cachedHash]) => {
+      console.log({ mainHash, cachedHash })
```
Comment (lmiller1990) [inline packages/electron/lib/install.js:62]: do we need this console log?

File: packages/electron/lib/install.js [hunk 3]
```diff
@@ -25,44 +23,66 @@ module.exports = {
...
+      const hashSum = crypto.createHash('sha1')
+
+      hashSum.update(buf)
+      const hash = hashSum.digest('hex')
+
+      return hash
+    })
+  },
+
+  checkIconVersion () {
```
Comment (lmiller1990) [inline packages/electron/lib/install.js:56]: Why do we need to update this? I see we are doing something with caching but why is this necessary just to change some icons?

File: packages/icons/lib/icons.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const path = require('path')
+const dist = [__dirname, '..', 'dist']
```
Comment (lmiller1990) [inline packages/icons/lib/icons.js:2]: Can we use TS for new packages?

You can just do this: https://github.com/cypress-io/cypress/blob/10.0-release/scripts/gulp/gulpfile.ts#L243

- `yarn gulp makePackage` And it will add a package with TS etc etc for you.

**Review by lmiller1990 — COMMENTED** [review]
Just a few comments, agree with @flotwig, why not make this `private` and not publish it on npm?

**Comment by elevatebart** [comment]
Yep, I did the change already

**Comment by lmiller1990** [comment]
Ok, pls make the TS updates @elevatebart

File: packages/electron/lib/install.js (hunk 3 above)
Comment (elevatebart) [reply packages/electron/lib/install.js:56]: On local, electron app is recompiled only if electron version changes or if the app is completely absent. 
If icon is updated, no contributor will see the new icons unless they delete the `dist` directory manually.

I compare the icon in dist with the one in the icons package and if different, force the re-build as well.

**Review by lmiller1990 — DISMISSED** [review]

File: packages/icons/readme.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
+# Cypress Icons
```
Comment (jennifer-shehane) [inline packages/icons/readme.md:1]: @elevatebart These readme instructions are out of date considering this is an internal package. Can you update the Readme to accurately reflect it's usage?

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
This logo doesn't look to be the correct logo, this used to be a back logo with transparent background.

[
Screen Shot 2022-02-25 at 2 27 23 PM
](url)

The favicon in the current tab is not displaying as the correct logo:

Screen Shot 2022-02-25 at 2 28 58 PM

**Comment by elevatebart** [comment]
Thank you @jennifer-shehane I will replace the light logo with a dark one.
Regarding the favicon, that seems to be a cache issue.

**Comment by elevatebart** [comment]
@jennifer-shehane 

I updated the icon with one that fits our design system.
I cache-busted the favicons so no one has to cache bust themselves.


The page referenced above is to be redesigned, just not in this story so don't worry too much about it ;)

https://cypress-io.atlassian.net/browse/UNIFY-1201

Thank you for the review

File: packages/electron/lib/install.js (hunk 2 above)
Comment (elevatebart) [reply packages/electron/lib/install.js]: no you are right good catch!
I removed it

File: packages/icons/readme.md (hunk 5 above)
Comment (elevatebart) [reply packages/icons/readme.md:1]: Fixed the readme, would appreciate a proof read.

File: packages/electron/lib/install.js (hunk 3 above)
Comment (flotwig) [reply packages/electron/lib/install.js:56]: This seems like a lot of code that has to be maintained for something that just requires devs to `git clean -xfd` once to bypass. I get why you added this, I think, but it's going to be dead code for 99.99% of runs and additional maintenance overhead, I think we could do without these `checkIconVersion` changes.

File: packages/app/index.html [hunk 6]
```diff
@@ -3,9 +3,9 @@
 
 <head>
   <meta charset="utf-8" />
-  <link href="/__cypress/assets/favicon.png" rel="icon">
+  <link href="/__cypress/assets/favicon.png?2022" rel="icon">
```
Comment (flotwig) [inline packages/app/index.html:6]: is this change necessary? i feel like this could use a comment explaining why there is a number here

File: packages/runner-ct/static/index.html [hunk 7]
```diff
@@ -7,7 +7,7 @@
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
       <title>{{projectName}}</title>
 
-      <link href="/{{namespace}}/runner/favicon.ico" rel="icon">
+      <link href="/{{namespace}}/runner/favicon.ico?2022" rel="icon">
```
Comment (flotwig) [inline packages/runner-ct/static/index.html:10]: ditto with previous comment, is this necessary? can it have a comment?

File: packages/runner/static/index.html [hunk 8]
```diff
@@ -5,7 +5,7 @@
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
       <title>{{projectName}}</title>
 
-      <link href="/{{namespace}}/runner/favicon.ico" rel="icon">
+      <link href="/{{namespace}}/runner/favicon.ico?2022" rel="icon">
```
Comment (flotwig) [inline packages/runner/static/index.html:8]: another 2021?

File: packages/server/lib/html/non_proxied_error.html [hunk 9]
```diff
@@ -4,7 +4,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Cypress</title>
 
-    <link href="/__cypress/runner/favicon.ico" rel="icon">
+    <link href="/__cypress/runner/favicon.ico?2022" rel="icon">
```
Comment (flotwig) [inline packages/server/lib/html/non_proxied_error.html:7]: 2022 now?

**Review by flotwig — CHANGES_REQUESTED** [review]

**Comment by flotwig** [comment]
Lookin good in Linux. For a split second, the Chrome favicon showed the old icon, but I guess that was from cache. 
image

**Comment by elevatebart** [comment]
@flotwig 
I do not disagree that it is a bit of code for cache busting
I am just thinking of future icons updaters.
literally, a few lines changed.

end-users will have the same problem with favicons

**Comment by flotwig** [comment]
>  literally, a few lines changed.

Complexity is complexity, is all. We change icons once every 7 years, but we regularly have to open this file and read and understand the code in there to fix bugs and add features. That's where I'm coming from, hope it helps.

> end-users will have the same problem with favicons

Aren't these changes to `@packages/electron` only applicable in `dev`, since we're comparing the built with the dev version?

File: packages/icons/readme.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+// => /Users/.../dist/icons/icon_32x32@2x.png
+```
+
+## Architecture detail
+
+To build the MacOS icons you have to use the `iconutil` command line tool installed.
+This command line tool is only installed on MacOS.
+
+If you are not on MacOS, the building of this icon will simply be skipped.
+Good news the macos is only needed to develop on MacOS.
```
Comment (lmiller1990) [inline packages/icons/readme.md:28]:
```suggestion
```

Comment (lmiller1990) [reply packages/icons/readme.md:28]: I don't think this line is adds any meaning.

File: packages/icons/readme.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+## Developing
+
+All icons are in the assets directory
```
Comment (lmiller1990) [inline packages/icons/readme.md:32]:
```suggestion
All the icons are in the assets directory.
```

File: packages/icons/readme.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+```bash
+## run build to dump to ./dist
+npm run build
```
Comment (lmiller1990) [inline packages/icons/readme.md:36]: `yarn` is used in all other READMEs and in the code base.

File: packages/icons/readme.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+The latest versions of the icons.
+The public API will always reference these files.
+
+`./dist` is not checked into source control.
+
+## API
+
+```js
+icons = require("@cypress/icons")
```
Comment (lmiller1990) [inline packages/icons/readme.md:11]:
```suggestion
const icons = require("@cypress/icons")
```

Comment (lmiller1990) [reply packages/icons/readme.md:11]: Nit, but nice to have valid code in the examples so you can copy paste.

File: packages/icons/package.json [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  "scripts": {
+    "build": "scripts/build.sh",
+    "postbuild": "ts-node ./src/ico.ts",
+    "test": "NODE_ENV=test mocha",
+    "test-watch": "NODE_ENV=test mocha --watch",
+    "prepublish": "npm run build",
+    "release": "releaser"
+  },
+  "devDependencies": {
+    "@types/mocha": "^9.1.0",
```
Comment (lmiller1990) [inline packages/icons/package.json:15]: Nit, can we use 8.x types for mocha - this is what most of our other packages reference. Can we match the types version?

**Review by lmiller1990 — COMMENTED** [review]
Sorry for the dragged out review, noticed some more things since you added some commits.

File: packages/icons/readme.md (hunk 10 above)
Comment (elevatebart) [reply packages/icons/readme.md:28]: It is meant to reassure the reader that even if they are not on macos, they will be able to build and work seemleassly.

How would you phrase that?

Comment (lmiller1990) [reply packages/icons/readme.md:28]: Right, I see. I think it's implied by "If you are not on MacOS, the building of this icon will simply be skipped.". This isn't really a PR blocker though.

File: packages/icons/package.json (hunk 14 above)
Comment (lmiller1990) [reply packages/icons/package.json:15]: @elevatebart can we change this one?

Comment (elevatebart) [reply packages/icons/package.json]: you got it !!

File: packages/icons/readme.md (hunk 13 above)
Comment (elevatebart) [reply packages/icons/readme.md]: agreed

File: packages/icons/readme.md (hunk 12 above)
Comment (elevatebart) [reply packages/icons/readme.md]: done

**Review by flotwig — APPROVED** [review]

**Review by ZachJW34 — APPROVED** [review]
